pkg/listers/executor/v1/webhooks.go

```diff
@@ -0,0 +1,101 @@
+*/
+
+package v1
+
+import (
+	executorv1 "github.com/kubeshop/testkube-operator/api/executor/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/cache"
+)
+
+// WebhookLister helps list Webhooks.
+// All objects returned here must be treated as read-only.
+type WebhookLister interface {
+	// List lists all Webhooks in the indexer.
+	// Objects returned here must be treated as read-only.
+	List(selector labels.Selector) (ret []*executorv1.Webhook, err error)
+	// Webhooks returns an object that can list and get Webhooks.
+	Webhooks(namespace string) WebhookNamespaceLister
+	WebhookListerExpansion
+}
+
+// webhookLister implements the WebhookLister interface.
+type webhookLister struct {
+	indexer cache.Indexer
+}
+
+// NewWebhookLister returns a new WebhookLister.
+func NewWebhookLister(indexer cache.Indexer) WebhookLister {
+	return &webhookLister{indexer: indexer}
+}
+
+// List lists all Webhooks in the indexer.
+func (s *webhookLister) List(selector labels.Selector) (ret []*executorv1.Webhook, err error) {
+	err = cache.ListAll(s.indexer, selector, func(m interface{}) {
+		ret = append(ret, m.(*executorv1.Webhook))
+	})
+	return ret, err
+}
+
+// Webhooks returns an object that can list and get Webhooks.
+func (s *webhookLister) Webhooks(namespace string) WebhookNamespaceLister {
+	return webhookNamespaceLister{indexer: s.indexer, namespace: namespace}
+}
+
+// WebhookNamespaceLister helps list and get Webhooks.
+// All objects returned here must be treated as read-only.
+type WebhookNamespaceLister interface {
+	// List lists all Webhooks in the indexer for a given namespace.
+	// Objects returned here must be treated as read-only.
+	List(selector labels.Selector) (ret []*executorv1.Webhook, err error)
+	// Get retrieves the Webhook from the indexer for a given namespace and name.
+	// Objects returned here must be treated as read-only.
+	Get(name string) (*executorv1.Webhook, error)
+	WebhookNamespaceListerExpansion
+}
+
+// webhookNamespaceLister implements the WebhookNamespaceLister
+// interface.
+type webhookNamespaceLister struct {
+	indexer   cache.Indexer
+	namespace string
+}
+
+// List lists all Webhooks in the indexer for a given namespace.
+func (s webhookNamespaceLister) List(selector labels.Selector) (ret []*executorv1.Webhook, err error) {
+	err = cache.ListAllByNamespace(s.indexer, s.namespace, selector, func(m interface{}) {
+		ret = append(ret, m.(*executorv1.Webhook))
+	})
+	return ret, err
+}
+
+// Get retrieves the Webhook from the indexer for a given namespace and name.
+func (s webhookNamespaceLister) Get(name string) (*executorv1.Webhook, error) {
+	obj, exists, err := s.indexer.GetByKey(s.namespace + "/" + name)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, errors.NewNotFound(
+			schema.GroupResource{Group: executorv1.GroupVersion.Group, Resource: executorv1.Resource},
```

Comment:
executorv1.Resource is Executor?

Reply:
Yes, I used what was in the groupversion_info.go of the package the Webhook struct is located in. I assume this could get confusing here, so I could either create an additional variable called `WebhookResource` in the groupversion, or do the cleaner way and extract Webhook into its own package in the `api` folder - but that could get into a more complex task. WDYT?

Reply:
yes, use ExecutorResource and WebhookResource

Reply:
Updated it, thanks!